humanlayer-wui/src/components/internal/SessionDetail/eventToDisplayObject.tsx

```diff
@@ -559,7 +560,16 @@ export function eventToDisplayObject(
               )}
             </>
           ) : (
-            <DenyButtons onCancel={onCancelDeny} isDenying={isDenying} />
+            <DenyButtons
+              onCancel={onCancelDeny}
+              onDeny={() => {
+                if (event.approvalId && onDeny) {
+                  onDeny(event.approvalId, responseText?.trim() || '')
+                }
+              }}
+              isDenying={isDenying}
+              isDisabled={!responseText?.trim()}
```

Comment:
This will work, but an alternate: We're calling `responseEditor.getText()` to check for whether the responseEditor has content, but `responseEditor` also has an `isEmpty` boolean we can check against that might be use-case appropriate for this.